File: switchyard/libsy/algorithms.py
# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

"""Factories for Rust-owned libsy algorithms."""

from switchyard_rust.libsy import llm_classifier as llm_classifier
from switchyard_rust.libsy import llm_task_classifier as llm_task_classifier
from switchyard_rust.libsy import noop as noop
from switchyard_rust.libsy import random as random
from switchyard_rust.libsy import stage_router as stage_router

__all__ = ["llm_classifier", "llm_task_classifier", "noop", "random", "stage_router"]

File: crates/libsy/src/algorithms.rs
// SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

//! Concrete algorithms and the interfaces for building them.
//!
//! Everything public here is re-exported at the crate root; reach for it by name —
//! `use switchyard_libsy::Random`.

pub mod advisor_gate;
pub mod composite;
mod escalation;
pub mod fall_through;
pub mod llm_class;
pub mod noop;
pub mod passthrough;
pub mod rand;
pub mod stage;
pub mod subagent;

pub mod util;

#[cfg(test)]
mod subagent_affinity_tests;
